client/src/client_sync/v21/wallet.rs

```diff
@@ -9,6 +9,66 @@
 //!
 //! See or use the `define_jsonrpc_minreq_client!` macro to define a `Client`.
 
+/// Implements Bitcoin Core JSON-RPC API method `createwallet` with descriptors=true (descriptor wallet)
+#[macro_export]
+macro_rules! impl_client_v21__create_wallet_with_descriptors {
+    () => {
+        impl Client {
+            pub fn create_wallet_with_descriptors(&self, wallet: &str) -> Result<CreateWallet> {
+                let args = [
+                    wallet.into(),
+                    false.into(),            // disable_private_keys
+                    false.into(),            // blank
+                    serde_json::Value::Null, // passphrase
+                    false.into(),            // avoid_reuse
+                    true.into(),             // descriptors=true
+                    serde_json::Value::Null, // load_on_startup
+                ];
+                self.call("createwallet", &args)
+            }
+        }
+    };
+}
```

Comment:
I think this macro should be done like the v23 one i.e., include function `create_wallet` as well as the with desrciptor function. Then in mod.rs of v23 and v24 only call this macro not the v17 one and from v23 onwards only call the v23 one.

Reply:
And I"d comment liberally about args to the RPC method changing and that we only include args we actually use in the integration tests - or some such thing.

Reply:
Addedd #292